lib/_http_client.js

```diff
@@ -22,6 +22,9 @@ function ClientRequest(options, cb) {
 
   if (typeof options === 'string') {
     options = url.parse(options);
+    if (!options.hostname) {
+      throw new Error('Unable to determine the domain name');
+    }
   } else {
     options = util._extend({}, options);
```

Comment:
@Fishrock123 At this place, if user doesn't pass `hostname`, then the default value `undefined` will be used